### ISSUE TITLE
Subscription Sink improve ergonomics of `send` API

### ISSUE THREAD

**Issue by lexnv** [paritytech/jsonrpsee#944]
The send API of the subscription sink can return the following:
- Ok(true): message was sent
- Ok(false): message not sent, because the client is disconnected
- Err(err): Error regarding serialization failure

https://github.com/paritytech/jsonrpsee/blob/dbac4bdb1d5223fee43856db9326e5fcde2de9bf/core/src/server/rpc_module.rs#L888-L902

This is a bit tricky to work with, given that users might try to match
only errors (ie `if let Err(err) = sink.send() { stop_service(); }`).

To improve the API, ensure that on success `Ok(())` is returned and on
failure have a dedicated error to differentiate between disconnect errors
or serde errors:

```rust
pub enum SubscriptionSinkError {
  ClientDisconnected,
  Serialization(serde::Error),
}
```

**Comment by niklasad1** [comment]
Ideally, folks don't have to bother with this API at all and use `pipe_from_sink`.

I think we designed it just like that around the `pipe_from_sink` i.e, not to regard `ConnectionClosed` as an error.
In addition `serde::Serialization errors` are most likely bugs and should not really occur.

Anyway, I'm all good with regarding all send errors as an error instead and to make this API cleaner.

We could also have some specific error type such as https://docs.rs/futures/latest/futures/channel/mpsc/struct.TrySendError.html#

**Comment by niklasad1** [comment]
Closed by #962
